tests/gui/TestGui.cpp

```diff
@@ -1473,7 +1477,96 @@ void TestGui::testDatabaseSettings()
     QTest::keyClick(transformRoundsSpinBox, Qt::Key_Enter);
     QTRY_COMPARE(m_db->kdf()->rounds(), 123456);
 
+    QVERIFY(autosaveDelayCheckBox->isChecked() == false);
+    autosaveDelayCheckBox->toggle();
+    autosaveDelaySpinBox->setValue(autosaveDelayTestValue);
+    QTest::mouseClick(dbSettingsButtonBox->button(QDialogButtonBox::Ok), Qt::LeftButton);
+    QTRY_COMPARE(m_db->metadata()->autosaveDelayMin(), autosaveDelayTestValue);
+    QApplication::processEvents();
+
     checkSaveDatabase();
+
+    fileDialog()->setNextFileName(m_dbFilePath);
+    triggerAction("actionDatabaseOpen");
```

Comment:
I had to use it to test all branches in `DatabaseSettingsWidgetGeneral::initialize()` because i was missing opening the settings as you noticed in previous comment you made at line 1480. This is now removed, will be pushed soon